lib/Yield/Support/GraphLayout/SugiyamaStyle/GraphPreparation.cpp

```diff
@@ -213,15 +220,15 @@ RankContainer partitionLongEdges(InternalGraph &Graph,
         revng_assert(Edge != nullptr);
         SavedLongEdges.emplace_back(From, To, std::move(*Edge));
         Iterator = From->removeSuccessor(Iterator);
+        if (!To->hasPredecessors())
+          GraphEntry->addSuccessor(To, InternalNode::Edge(Graph));
```

Comment:
So here we've removed a long edge that was going into `To`, and `To` might become orphan. If it does, it becomes unreachable from the entry point, so we're adding a placeholder edge from the virtual entry point to `To`, so that it remains reachable.

Intuitively it makes sense, but I have a question: at the end of `partitionLongEdges` the virtual entry guard goes out of scope, so it deletes the virtual entry point, along with all the outgoing edges from it. At that point `To` would become orphan again. Can't this cause problems down the line?

Reply:
Every top-corner-style V-shape is an orphan by definition. So it is just going to be one of many.

This is the entire point of going with the guard approach - we only care about a clean entry point at limited points in the algorithm (and only because LLVM does), so if we just wrap these limited points in the guard - everyone is happy - and it's easy to see where we are allowed to rely on that (because there's a guard in the scope), and where we can't.

Am I missing something?